catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/dialog/ReplaceApiKeyDialogTest.java

```diff
@@ -0,0 +1,139 @@
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.ui.dialog;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import com.google.common.base.Charsets;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.Constants;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.StartScript;
+import org.catrobat.catroid.content.bricks.BackgroundRequestBrick;
+import org.catrobat.catroid.io.asynctask.ProjectSaveTask;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.uiespresso.ui.activity.ProjectUploadRatingDialogTest;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityTestRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+import androidx.test.core.app.ApplicationProvider;
+
+import static org.catrobat.catroid.common.Constants.CODE_XML_FILE_NAME;
+import static org.catrobat.catroid.common.SharedPreferenceKeys.AGREED_TO_PRIVACY_POLICY_VERSION;
+import static org.catrobat.catroid.ui.ProjectUploadActivity.PROJECT_DIR;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+public class ReplaceApiKeyDialogTest {
+
+	private static final String TAG = ReplaceApiKeyDialogTest.class.getSimpleName();
+
+	@Rule
+	public BaseActivityTestRule<ProjectUploadRatingDialogTest.ProjectUploadTestActivity> activityTestRule =
+			new BaseActivityTestRule<>(ProjectUploadRatingDialogTest.ProjectUploadTestActivity.class, false, false);
+
+	private int bufferedPrivacyPolicyPreferenceSetting;
+
+	Project dummyProject;
+	String apikey = "AIzaas98d7f9a0sdf07ad0sf8a7sd09fASDf97asd9f";
+	@Before
+	public void before() {
+		TestUtils.deleteAllProjects();
```

Comment:
please remove this. `BaseActivityTestRule` is taking care of that during the initialization. look at its implementation.